libs/commands/publish/src/lib/throttle-queue.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * A queue abstraction for tasks that may require a delayed execution
+ */
+export interface Queue {
+  /**
+   * Run a task once the queue is cleared
+   * @param f A function that will be executed once the queue is useable
+   * @return A promise that wraps the value returned by f
+   */
+  queue<T>(f: () => Promise<T>): Promise<T>;
+}
+
+/**
+ * A queue that immediately executes any promise it receives
+ */
+export class ImmediateQueue implements Queue {
+  queue<T>(f: () => Promise<T>): Promise<T> {
+    return f();
+  }
+}
+
+/**
+ * A sized queue that adds a delay between the end of an item's execution.
+ *
+ * Use only with async code, multi-threading is not supported.
+ */
+export class TailHeadQueue implements Queue {
+  private queue_list: ((v: any) => any)[];
+  private queue_size: number;
+  private queue_period: number;
+  private allowance: number;
+  private last_end: number[];
+
+  /**
+   * @param size The number of items that may run concurrently
+   * @param period The time between the end of the execution of an item and the start of the execution of the next one (ms)
+   */
+  constructor(size = 20, period = 30 * 1000) {
```

Comment:
I'd prefer this class to require this data to be passed in and only have a set of default values defined once for maximum clarity